src/qibolab/result.py

```diff
@@ -108,7 +108,7 @@ class SampleResults:
     """
 
     def __init__(self, data: np.ndarray):
-        self.samples: npt.NDArray[np.uint32] = data
+        self.samples: npt.NDArray[np.uint32] = np.array(data).astype(int)
```

Comment:
I believe `int` could be platform dependent, and possibly is `np.int64`, despite usually being 32 bit for the C compiler. 
```suggestion
        self.samples: npt.NDArray[np.uint32] = np.array(data).astype(np.uint32)
```

It depends if it is interpreted as `np.intc` or `np.int_`.
https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.intc
https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.int_

Reply:
It's `np.int_` (the `_` was much more likely than one letter more, it would have been quite arbitrary)
```py
[ins] In [4]: a = np.arange(10, dtype=int)
                                                                                                                                                                                                                                                                                                                                                                            
[ins] In [5]: a
Out[5]: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
                                                                                                                                                                                                                                                                                                                                                                            
[ins] In [6]: a.dtype
Out[6]: dtype('int64')
```